tendermint/src/chain/id.rs

```diff
@@ -2,89 +2,100 @@
 
 use crate::error::{Error, Kind};
 use serde::{de::Error as _, Deserialize, Deserializer, Serialize, Serializer};
+use std::convert::TryFrom;
 use std::{
     cmp::Ordering,
     fmt::{self, Debug, Display},
     hash::{Hash, Hasher},
     str::{self, FromStr},
 };
+use tendermint_proto::DomainType;
 
 /// Maximum length of a `chain::Id` name. Matches `MaxChainIDLen` from:
 /// <https://github.com/tendermint/tendermint/blob/develop/types/genesis.go>
 // TODO: update this when `chain::Id` is derived from a digest output
 pub const MAX_LENGTH: usize = 50;
 
 /// Chain identifier (e.g. 'gaia-9000')
-#[derive(Copy, Clone)]
-pub struct Id([u8; MAX_LENGTH]);
+#[derive(Clone)]
```

Comment:
```suggestion
#[derive(Copy, Clone)]
```

Reply:
This, unfortunately won't work, because the implemented `String` does not have the `Copy` trait. If you need bytes, you can use `as_bytes()` or `as_str()`. (You can read about the issue with the Copy trait and String in the Rust book. It's a size-in-memory thing.)

Reply:
Alternatively, we can revert back to using [u8] internally but it seemed to me that `String` is a better representation with the added functions. Up to the team to decide.